auth/src/main/java/org/aerogear/mobile/security/AbstractSecurityCheckExecutor.java

```diff
@@ -5,76 +5,82 @@
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
+import org.aerogear.mobile.core.executor.AppExecutors;
 import org.aerogear.mobile.core.metrics.MetricsService;
 
 import java.util.Collection;
 import java.util.HashSet;
 
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
 /**
  * Base class for security check executors.
  */
 abstract class AbstractSecurityCheckExecutor<T extends AbstractSecurityCheckExecutor> {
 
-    /**
-     * Collection of checks to be executed.
-     */
     private final Collection<SecurityCheck> checks = new HashSet<>();
-
-    /**
-     * Context.
-     */
     private final Context context;
-
-    /**
-     * Metric singleThreadService to be used to publish metrics. Can be null.
-     */
     private final MetricsService metricsService;
 
     /**
-     * Constructor.
+     * Creates AbstractSecurityCheckExecutor object.
      *
-     * @param context the context
-     * @param checks checks to be executed
-     * @param metricService singleThreadService to be used to publish metrics. If null, no metrics get published.
+     * @param context {@link Context} to be used by the security checks
+     * @param checks  {@link Collection} of {@link SecurityCheck} to be executed
+     * @param metricService {@link MetricsService}, which should be a {@link AppExecutors#singleThreadService()}, to be used to publish metrics. If null, no metrics get published
+     * @throws IllegalArgumentException if context or checks are null
      */
     public AbstractSecurityCheckExecutor(@NonNull final Context context,
                                          @NonNull final Collection<SecurityCheck> checks,
                                          @Nullable final MetricsService metricService) {
-        this.context = context;
-        this.metricsService = metricService;
+        this.context = nonNull(context, "context");
         this.checks.addAll(checks);
+        this.metricsService = metricService;
     }
 
     /**
      * Gets all the checks to be executed.
-     * @return all the checks to be executed
+     *
+     * @return {@link Collection} of {@link SecurityCheck}
      */
     protected Collection<SecurityCheck> getChecks() {
         return checks;
     }
 
+    /**
+     * Gets the context for the device.
+     *
+     * @return {@link Context}
+     */
     protected Context getContext() {
         return context;
     }
 
+    /**
+     * Gets the metrics service.
+     *
+     * @return {@link MetricsService}
+     */
     protected MetricsService getMetricsService() {
         return metricsService;
     }
 
     /**
-     * Adds a new check to be executed
-     * @param check the new check to be executed
-     * @return this, so that adding checks can be chained
+     * Adds a new check to be executed.
+     *
+     * @param check the new {@link SecurityCheck} to be executed
+     * @return {@link T} this, so that adding checks can be chained
```

Comment:
I don't think this works. Have you tried generating the Javadocs? T is a generic type, I don't think javadoc will be able to generate any link to T.